src/components/Calculator.vue

```diff
@@ -30,22 +30,149 @@ export default {
   name: "Calculator",
   data() {
     return {
-      output: "123"
-    }
-  }, 
+      display: "0",
+      output: "0",
+      baseNumber: 0,
+      newNumber: 0,
+      operator: null,
+      equalBtnPressed: false,
+      selectedOperator: "",
+    };
+  },
   methods: {
-    clear() {
-      this.output = ""
-    }
-  }
+    updateDisplay() {
+      this.display = this.output;
+    },
+    clearBtn() {
+      this.display = "0";
+      this.output = "0";
+      this.baseNumber = 0;
+      this.newNumber = 0;
+      this.operator = null;
+      this.equalBtnPressed = false;
+      this.selectedOperator = "0;";
+    },
+    appendBtn(value) {
+      this.selectedOperator = "";
+      if (this.output.charAt(0) === "0" && this.output.charAt(1) !== ".") {
+        this.output = "";
+      }
+      this.output += value;
+      this.updateDisplay();
+    },
+    appendDotBtn() {
+      this.selectedOperator = "";
+      if (this.output.indexOf(".") < 0) {
+        this.output += ".";
+      }
+      this.updateDisplay();
+    },
+    negateBtn() {
+      if (this.output !== "0") {
+        let value = parseFloat(this.output) * -1;
+        this.output = "" + value;
+        this.updateDisplay();
+      }
+    },
+    percentageBtn() {
+      this.output = "" + parseFloat(this.output) / 100;
+    },
+    additionBtn() {
+      if (this.selectedOperator === "addition") return;
+      this.equalBtnPressed = false;
+      this.newNumber = parseFloat(this.output);
+      this.operator = (a, b) => a + b;
+      this.output = this.operator(this.baseNumber, this.newNumber);
+      this.baseNumber = this.output;
+      this.updateDisplay();
+      this.output = "0";
+      this.selectedOperator = "addition";
+    },
+    equalBtn() {
+      if (this.selectedOperator !== "") {
+        this.newNumber = this.baseNumber;
+      }
+      if (!this.equalBtnPressed && this.selectedOperator == "") {
+        this.newNumber = parseFloat(this.output);
+        this.equalBtnPressed = true;
+      }
+      this.output = this.operator(this.baseNumber, this.newNumber);
+      this.baseNumber = this.output;
+      this.updateDisplay();
+      this.output = "0";
+    },
+    subtractBtn() {
+      if (this.selectedOperator === "subtraction") return;
```

Comment:
It looks here like you are assigning the operation to happen, and then carrying it out straight away? Also it isn't doing any operation if you have already done a subtraction?

It would be good to know if you intend to have the calculator only perform the equation when hitting "=" or on each click of an operation, either way, the current functions are neither.

Reply:
I went for the behaviour of when pressing the operation buttons(+,-,/,*) will continue doing the calculation, so I went for the approach of setting the operation and also performing the calculation within each operation button. 

Each operation and equal button will capture whatever number has been entered on the display and then performs the calculation by either adding, subtracting, multiplying, or dividing the baseNumber to the captured newNumber.

Reply:
That makes sense, but if the selected operator is clicked twice, like 5 - 2 - 1 it wont do the second one because the selected operator is already -?

Reply:
I manage to get it to do continuous calculation even if the '=' operator has not been pressed, so the example 5 - 2 -1 would still work. 

Each time when an operator is pressed, the 'operator' gets set to  a function that does the corresponding arithmetic.
The function also gets called which uses the current number 'output' and the 'baseNumber'. At the end of the method, the 'selectOperator' is also set to either "addition", "subtraction", "division", or "multiplication" depending on the operator. This is to stop the issue of continuously pressing the same operator button that will continuously perform the calculation (e.g.  5+5+++ will get us 25). 
It also allow us to switch the operator if we accidentally hit the wrong operator (e.g. 5-1-2-+2 will get us 4) . 

When any digit button or decimal button has been pressed, the 'selectOperator' changes to "" which will allow us to select the same or different operator.